src/Tokenizer/TokensAnalyzer.php

```diff
@@ -564,6 +564,10 @@ public function isBinaryOperator(int $index): bool
         $tokens = $this->tokens;
         $token = $tokens[$index];
 
+        if (isset($potentialUnaryNonArrayOperators[$token->getContent()])) {
```

Comment:
nice find, but sadly this won't work; https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6089

Reply:
it will, with `if ($token->isGivenKind(T_ENCAPSED_AND_WHITESPACE))`, although I'm open for better solution